uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/API/Query/LeaveRequestSelect.php

```diff
@@ -141,9 +154,9 @@ private function addJoins(CRM_Utils_SQL_Select $query) {
       'INNER JOIN ' . HRJobDetails::getTableName() . ' jd ON jd.jobcontract_revision_id = jcr.details_revision_id'
     ];
 
-    if(!empty($this->params['managed_by'])) {
-      $joins[] = 'INNER JOIN ' . Relationship::getTableName() . ' r ON r.contact_id_a = a.contact_id';
-      $joins[] = 'INNER JOIN ' . RelationshipType::getTableName() . ' rt ON rt.id = r.relationship_type_id';
+    if(!empty($this->params['managed_by']) || isset($this->params['unassigned'])) {
+      $joins[] = 'LEFT JOIN ' . Relationship::getTableName() . ' r ON r.contact_id_a = a.contact_id';
```

Comment:
I might be wrong here because I didn't expect the generated queries, but I have the impression that, since now this is a LEFT JOIN, the whole logic can be simplified:

- For unassigned = true: We filter out all the records where r.id IS NOT NULL (That is, those that have one relationship)
- For unassigned = false: We filter out all the records where r.id IS NULL (That is, those that don't have one relationship)
- For when unassigned is not present: Don't do anything. The query should just work like before.